File: TP_webApp/TP_webApp/Model/RefMessage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace TP_webApp.Model
{
    public class RefMessage : Message
    {
        [JsonIgnore]
        private Message _message = null;
        [JsonIgnore]
        public Message message { get {
            if (_message == null) {
                _message = this.parentDiagram.layers[layerIndex].messages[messageIndex];
            }
            return _message;
        } set {
            message = value;
        } }

        public int layerIndex { get; set; }
        public int messageIndex { get; set; }
        [JsonIgnore]
        public Diagram parentDiagram { get; set; }

        public RefMessage(Message message): base(null, kinds.unknow, sorts.synchCall, null, null) {
            this.message = message;
        }
        
        [JsonIgnore]
        public new kinds kind { get {
            return message.kind;
        } set {
            message.kind = value;
        } }

        [JsonIgnore]
        public new sorts sort { get {
            return message.sort;
        } set {
            message.sort = value;
        } }

        [JsonIgnore]
        public new String name { get {
            return message.name;
        } set {
            message.name = value;
        } }

        [JsonIgnore]
        public new Lifeline start { get {
            return message.start;
        } set {
            message.start = value;
        } }

        [JsonIgnore]
        public new Lifeline end { get {
            return message.end;
        } set {
            message.end = value;
        } }


        
    }
}

File: TP_webApp/TP_webApp/app/model/RefMessage.ts
import { Message, sorts, kinds } from "./Message";
import { OccurenceSpecification } from "./OccurenceSpecification";
import { Layer } from "./Layer";
import { GraphicElement } from "../view/GraphicElement";

export class RefMessage extends Message {

    private message: Message;

    private layerIndex: number;
    private messageIndex: number;

    public get name(): string {
        return this.message._name;
    }

    public set name(name: string) {
        this.message._name = name;
    }

    public get sort(): sorts {
        return this.message._sort;
    }

    public set sort(sort: sorts) {
        this.message._sort = sort;
    }

    public get kind(): kinds {
        return this.message._kind;
    }

    public set kind(kind: kinds) {
        this.message._kind = kind;
    }

    public get start(): OccurenceSpecification {
        return this.message._start;
    }

    public set start(start: OccurenceSpecification) {
        this.message._start = start;
    }

    public get end(): OccurenceSpecification {
        return this.message._end;
    }

    public set end(end: OccurenceSpecification) {
        this.message._end = end;
    }

    public get parentLayer(): Layer {
        return this.start.at.layer;
    }

    public get graphicElement(): GraphicElement {
        return this.message.graphicElement;
    }

    public set graphicElement(graphicElement: GraphicElement) {
        this.message.graphicElement = graphicElement;
    }

    constructor(message: Message) {
        super(null, null, null, null, null);
        this.message = message;
    }

}
